### PR TITLE
Explicit `RETURNING` for `InsertStatement` and `UpdateStatement`

### LINKED ISSUE #180: Add `returning` to `InsertStatement` and `UpdateStatement`
The ability for `RETURNING *` to automatically get added is *super* useful, but sometimes we only want one or two fields. We should allow the user to be explicit about what they want back here.

### PR DESCRIPTION
**Pull request by robertmaloney**
Fixes #180.

### CODE REVIEW

File: diesel/src/query_builder/insert_statement.rs [hunk 1]
```diff
@@ -68,6 +68,18 @@ impl<T, U> AsQuery for InsertStatement<T, U> where
     }
 }
 
+impl<T, U> InsertStatement<T, U> {
+    /// Specify what expression is returned after execution of the `insert`.
```
Comment (sgrif) [inline diesel/src/query_builder/insert_statement.rs]: Can you add a usage example?

File: diesel/src/query_builder/update_statement/mod.rs [hunk 2]
```diff
@@ -59,35 +59,53 @@ impl<T, U, DB> QueryFragment<DB> for UpdateStatement<T, U> where
...
-        UpdateQuery(self)
+        UpdateQuery {
+            returning: T::Table::all_columns(),
+            statement: self,
+        }
+    }
+}
+
+impl<T, U> UpdateStatement<T, U> {
+    /// Specify what expression is returned after execution of the `update`.
```
Comment (sgrif) [inline diesel/src/query_builder/update_statement/mod.rs]: Can you add a usage example?

File: diesel/src/query_builder/update_statement/mod.rs [hunk 3]
```diff
@@ -59,35 +59,53 @@ impl<T, U, DB> QueryFragment<DB> for UpdateStatement<T, U> where
...
+    pub fn returning<E>(self, returns: E) -> UpdateQuery<E, UpdateStatement<T, U>> where
+        E: Expression,
```
Comment (sgrif) [inline diesel/src/query_builder/update_statement/mod.rs]: This needs to check `SelectableExpression`, as well.

File: diesel/src/query_builder/insert_statement.rs [hunk 4]
```diff
@@ -68,6 +68,18 @@ impl<T, U> AsQuery for InsertStatement<T, U> where
...
+    pub fn returning<E>(self, returns: E) -> InsertQuery<E, InsertStatement<T, U>> where
+        E: Expression,
```
Comment (sgrif) [inline diesel/src/query_builder/insert_statement.rs]: This needs to check `SelectableExpression`, as well. Or more likely check that `InsertQuery<E, InsertStatement<T, U>>: Query`, and add the `SelectableExpression` constraint there.

File: diesel/src/query_builder/update_statement/mod.rs [hunk 5]
```diff
@@ -59,35 +59,53 @@ impl<T, U, DB> QueryFragment<DB> for UpdateStatement<T, U> where
...
         out.push_sql(" RETURNING ");
-        try!(T::Table::all_columns().to_sql(out));
+        try!(self.returning.to_sql(out));
         Ok(())
     }
 }
 
 impl<T, U> Query for UpdateQuery<T, U> where
-    T: UpdateTarget,
+    T: Expression,
```
Comment (sgrif) [inline diesel/src/query_builder/update_statement/mod.rs]: This removes a lot of type safety constraints. This needs to continue to verify that we're referencing an update target.

**Comment by sgrif** [comment]
This needs compile fail tests for all of the following:

- `update(users).set(changes).returning(posts::title)` needs to fail (columns from other tables, or more specifically, violating `SelectableExpression`)
- `update(users).set(changes).returning(count(id))` needs to fail (cannot have aggregate expressions in a returning clause)
- Attempting to run a query using an explicit call to `returning` on SQLite.

File: diesel/src/query_builder/update_statement/mod.rs (hunk 2 above)
Comment (robertmaloney) [inline diesel/src/query_builder/update_statement/mod.rs]: Done

File: diesel/src/query_builder/insert_statement.rs (hunk 1 above)
Comment (robertmaloney) [inline diesel/src/query_builder/insert_statement.rs]: Done

**Comment by robertmaloney** [comment]
Current comments have been addressed.

**Pull request merged**